packages/playground/src/pages/tools/tool/index.tsx

```diff
@@ -10,26 +11,78 @@ import {
   Button,
   DocsIcon,
   ToolPanel,
+  HeaderGroup,
+  Combobox,
+  useTools,
+  useAgents,
+  useLinkComponent,
 } from '@mastra/playground-ui';
 
 const Tool = () => {
   const { toolId } = useParams();
+  const { navigate, paths } = useLinkComponent();
+  const { data: tools = {} } = useTools();
+  const { data: agents = {} } = useAgents();
+
+  const toolOptions = useMemo(() => {
```

Comment:
we can get rid off the useMemo everywhere, those components are not rendering enough to be annoying